crates/sui-json-rpc-tests/tests/name_service_tests.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+use std::time::SystemTime;
+use sui_json_rpc::name_service;
+use sui_types::{
+    base_types::{ObjectID, SuiAddress},
+    collection_types::VecMap,
+};
+
+#[test]
+fn test_parent_extraction() {
+    let mut name = name_service::Domain::from_str("leaf.node.test.sui").unwrap();
+
+    assert_eq!(name.parent().to_string(), "node.test.sui");
+
+    name = name_service::Domain::from_str("node.test.sui").unwrap();
+
+    assert_eq!(name.parent().to_string(), "test.sui");
+}
+
+#[test]
+fn test_sld_extraction() {
+    let name = name_service::Domain::from_str("nested.leaf.node.test.sui").unwrap();
+
+    assert_eq!(name.sld().to_string(), "test.sui");
+}
+
+#[test]
+fn test_expirations() {
+    let expiration = SystemTime::now()
```

Comment:
It's not doing much harm here, but seeing as you don't need it, I would just remove the dependency on actual time here and come up with some fake `u64` to put in here.